### PR TITLE
fix(opentrons-ai-client): add conditional for user prompt

### PR DESCRIPTION
**Pull request by koji**
# Overview
add conditional for user prompt

### before
Screenshot 2024-05-31 at 3 36 48 PM

### after

Screenshot 2024-05-31 at 3 50 45 PM

close AUTH-433

# Test Plan

# Changelog

# Review requests

# Risk assessment

### CODE REVIEW

**Review by shlokamin — CHANGES_REQUESTED** [review]
i get this error from react markdown when using one of the example buttons: 

```
Warning: React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

```

Screenshot 2024-05-31 at 4 34 32 PM

**Comment by shlokamin** [comment]
i don't think react markdown likes `undefined`

**Comment by koji** [comment]
> i don't think react markdown likes `undefined`

change `undefined` to paragraph text.

File: opentrons-ai-client/src/molecules/ChatDisplay/index.tsx [hunk 1]
```diff
@@ -72,12 +72,13 @@ export function ChatDisplay({ chat, chatId }: ChatDisplayProps): JSX.Element {
             h2: HeaderText,
             li: ListItemText,
             p: ParagraphText,
-            a: ExternalLink,
-            code: CodeText,
+            a: isUser ? ParagraphText : ExternalLink,
+            code: isUser ? ParagraphText : CodeText,
```
Comment (Elyorcv) [inline opentrons-ai-client/src/molecules/ChatDisplay/index.tsx:76]: If I understand correctly, only `user` needs to be wrapped.

But, we need to wrap both `user` and `assistant` with markdown.

Comment (koji) [reply opentrons-ai-client/src/molecules/ChatDisplay/index.tsx:76]: what do you mean `wrapped`?

**Comment by Elyorcv** [comment]
> > i don't think react markdown likes `undefined`
> 
> change `undefined` to paragraph text.

does it behave differently for `user` vs. `assistant`?

**Comment by koji** [comment]
> > > i don't think react markdown likes `undefined`
> > 
> > 
> > change `undefined` to paragraph text.
> 
> does it behave differently for `user` vs. `assistant`?

yes because of conditional rendering.

**Review by Elyorcv — CHANGES_REQUESTED** [review]
No it is not behaving correctly. See the image below, when user sends info with backtick, it renders as a block.

```text

Write a protocol using the Opentrons Python Protocol API v2 for OT-2 robot according to the following description:
  
Labware:
- Source Labware: `Thermo Scientific Nunc 96 Well Plate 1300 uL` in slot 9
- Destination Labware: `Opentrons 96 Flat Bottom Adapter with NEST 96 Well Plate 200 uL Flat` in slot 10
- `Opentrons 96 Filter Tip Rack 1000 uL` in slot 8
- `Opentrons 96 Tip Rack 1000 uL` in slot 3

Pipette mount:
- P1000 Single-Channel GEN2 is mounted on the left
- P1000 Single-Channel GEN2 is mounted on the right

Commands:
1. Use the left-mounted P1000 Single-Channel GEN2 pipette to transfer 196 uL of reagent from wells A7, A6, A5, A2, A3 
of the source labware to the corresponding wells A5, A9, A1, A10, A2 of the destination labware. Use a new tip for each transfer.
2. Use the right-mounted P1000 Single-Channel GEN2 pipette to transfer 8 uL of liquid from wells A9, A12, A6, A10, A3 
of the source labware to the corresponding wells A7, A11, A6, A3, A9 of the destination labware. Use the same tip for all transfers.

```

image

**Comment by koji** [comment]
> No it is not behaving correctly. See the image below, when user sends info with backtick, it renders as a block.
> 
> ```
> 
> Write a protocol using the Opentrons Python Protocol API v2 for OT-2 robot according to the following description:
>   
> Labware:
> - Source Labware: `Thermo Scientific Nunc 96 Well Plate 1300 uL` in slot 9
> - Destination Labware: `Opentrons 96 Flat Bottom Adapter with NEST 96 Well Plate 200 uL Flat` in slot 10
> - `Opentrons 96 Filter Tip Rack 1000 uL` in slot 8
> - `Opentrons 96 Tip Rack 1000 uL` in slot 3
> 
> Pipette mount:
> - P1000 Single-Channel GEN2 is mounted on the left
> - P1000 Single-Channel GEN2 is mounted on the right
> 
> Commands:
> 1. Use the left-mounted P1000 Single-Channel GEN2 pipette to transfer 196 uL of reagent from wells A7, A6, A5, A2, A3 
> of the source labware to the corresponding wells A5, A9, A1, A10, A2 of the destination labware. Use a new tip for each transfer.
> 2. Use the right-mounted P1000 Single-Channel GEN2 pipette to transfer 8 uL of liquid from wells A9, A12, A6, A10, A3 
> of the source labware to the corresponding wells A7, A11, A6, A3, A9 of the destination labware. Use the same tip for all transfers.
> ```
> 
> image

not completely align with that since we use react-mark down and it take some time.
this pr is to fix the margin and no black background in user prompt.

https://opentrons.slack.com/archives/C074KKV6SRK/p1717189125167839

**Comment by Elyorcv** [comment]
> > No it is not behaving correctly. See the image below, when user sends info with backtick, it renders as a block.
> > ```
> > 
> > Write a protocol using the Opentrons Python Protocol API v2 for OT-2 robot according to the following description:
> >   
> > Labware:
> > - Source Labware: `Thermo Scientific Nunc 96 Well Plate 1300 uL` in slot 9
> > - Destination Labware: `Opentrons 96 Flat Bottom Adapter with NEST 96 Well Plate 200 uL Flat` in slot 10
> > - `Opentrons 96 Filter Tip Rack 1000 uL` in slot 8
> > - `Opentrons 96 Tip Rack 1000 uL` in slot 3
> > 
> > Pipette mount:
> > - P1000 Single-Channel GEN2 is mounted on the left
> > - P1000 Single-Channel GEN2 is mounted on the right
> > 
> > Commands:
> > 1. Use the left-mounted P1000 Single-Channel GEN2 pipette to transfer 196 uL of reagent from wells A7, A6, A5, A2, A3 
> > of the source labware to the corresponding wells A5, A9, A1, A10, A2 of the destination labware. Use a new tip for each transfer.
> > 2. Use the right-mounted P1000 Single-Channel GEN2 pipette to transfer 8 uL of liquid from wells A9, A12, A6, A10, A3 
> > of the source labware to the corresponding wells A7, A11, A6, A3, A9 of the destination labware. Use the same tip for all transfers.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > image
> 
> not completely align with that since we use react-mark down and it take some time. this pr is to fix the margin and no black background in user prompt.
> 
> https://opentrons.slack.com/archives/C074KKV6SRK/p1717189125167839

black background is coming from rendering of backtick. Although it is not black anymore, the backtick is still considered as a block rather than inline. Maybe we need to change this PR to account for only fixing margin.

**Comment by koji** [comment]
> > > No it is not behaving correctly. See the image below, when user sends info with backtick, it renders as a block.
> > > ```
> > > 
> > > Write a protocol using the Opentrons Python Protocol API v2 for OT-2 robot according to the following description:
> > >   
> > > Labware:
> > > - Source Labware: `Thermo Scientific Nunc 96 Well Plate 1300 uL` in slot 9
> > > - Destination Labware: `Opentrons 96 Flat Bottom Adapter with NEST 96 Well Plate 200 uL Flat` in slot 10
> > > - `Opentrons 96 Filter Tip Rack 1000 uL` in slot 8
> > > - `Opentrons 96 Tip Rack 1000 uL` in slot 3
> > > 
> > > Pipette mount:
> > > - P1000 Single-Channel GEN2 is mounted on the left
> > > - P1000 Single-Channel GEN2 is mounted on the right
> > > 
> > > Commands:
> > > 1. Use the left-mounted P1000 Single-Channel GEN2 pipette to transfer 196 uL of reagent from wells A7, A6, A5, A2, A3 
> > > of the source labware to the corresponding wells A5, A9, A1, A10, A2 of the destination labware. Use a new tip for each transfer.
> > > 2. Use the right-mounted P1000 Single-Channel GEN2 pipette to transfer 8 uL of liquid from wells A9, A12, A6, A10, A3 
> > > of the source labware to the corresponding wells A7, A11, A6, A3, A9 of the destination labware. Use the same tip for all transfers.
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > image
> > 
> > 
> > not completely align with that since we use react-mark down and it take some time. this pr is to fix the margin and no black background in user prompt.
> > https://opentrons.slack.com/archives/C074KKV6SRK/p1717189125167839
> 
> black background is coming from rendering of backtick. Although it is not black anymore, the backtick is still considered as a block rather than inline. Maybe we need to change this PR to account for only fixing margin.

because the response from backend includes pre tag(backtick)

What this pr does
- adding padding to li (this is obvious when the li item's number is more than 10 attached the image in overvew)
- not displaying black background in user prompt

https://github.com/Opentrons/opentrons/assets/474225/9e3b562f-cac4-4bd8-ae8c-3dfb1e09b3c9

**Review by Elyorcv — APPROVED** [review]
LGTM, I created another ticket to solve `backtick` rendering https://opentrons.atlassian.net/browse/AUTH-465

**Review by shlokamin — APPROVED** [review]

**Pull request merged**